test/Fencer/Rules/Test.hs

```diff
@@ -26,20 +27,58 @@ tests = testGroup "Rule tests"
   [ test_rulesLoadRulesYaml
   , test_rulesLoadRulesNonYaml
   , test_rulesLoadRulesRecursively
+  , test_rulesLoadRulesDotDirectory
+  , test_rulesLoadRulesRUNTIME_IGNOREDOTFILES
   ]
 
+-- | A helper function for loading rules and making sure they are as
+-- expected.
+loadRules :: String -> Bool -> IO ()
+loadRules dirTemplate ignoreDotFiles =
+  Temp.withSystemTempDirectory dirTemplate $ \tempDir -> do
+    TIO.writeFile (tempDir </> "config1.yml") domain1Text
+    TIO.writeFile (tempDir </> "config2.yaml") domain2Text
+    definitions <-
+      loadRulesFromDirectory
+        (#directory tempDir)
+        (#ignoreDotFiles ignoreDotFiles)
+    assertEqual "unexpected definitions"
+      (sortOn domainDefinitionId [domain1, domain2])
+      (sortOn domainDefinitionId definitions)
+
+
 -- | test that 'loadRulesFromDirectory' loads rules from YAML files.
 test_rulesLoadRulesYaml :: TestTree
 test_rulesLoadRulesYaml =
   testCase "Rules are loaded from YAML files" $
+    loadRules "fencer-config" True
+
+-- | test that 'loadRulesFromDirectory' loads rules from a
+-- dot-directory when dot-files should be ignored.
+test_rulesLoadRulesDotDirectory :: TestTree
+test_rulesLoadRulesDotDirectory =
+  testCase "Rules are loaded from a dot-directory" $
+    loadRules ".fencer-config" True
+
+
+-- | test that 'loadRulesFromDirectory' respects the
+-- RUNTIME_IGNOREDOTFILES environment variable.
+test_rulesLoadRulesRUNTIME_IGNOREDOTFILES :: TestTree
+test_rulesLoadRulesRUNTIME_IGNOREDOTFILES =
```

Comment:
This test essentially checks that `getSettingsFromEnvironment` works. This isn't much.

I think that we should just have two tests for `loadRulesFromDirectory ... (#ignoreDotFiles True)`:

```
-- test that dot-directories are not ignored
<tmpdir>
    .dir1
        domain1.yml
```

```
-- test that dot-files are ignored
<tmpdir>
    dir2
        .domain2.yml
```

If we want to check that `getSettingsFromEnvironment` uses right variable names, we can do it separately (though I think we don't have to do this at all).

Reply:
Addressed! I got rid of `getSettingsFromEnvironment` in the test.